dask/array/creation.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import absolute_import
+from itertools import count
+
+import numpy as np
+from toolz import curry
+
+from .core import Array
+
+linspace_names = ('linspace-%d' % i for i in count(1))
+arange_names = ('arange-%d' % i for i in count(1))
+
+
+def _get_blocksizes(num, blocksize):
+    # compute blockdims
+    remainder = (num % blocksize,)
+    if remainder == (0,):
+        remainder = tuple()
+    blocksizes = ((blocksize,) * (num // blocksize) + remainder)
+    return blocksizes
+
+
+def linspace(start, stop, blocksize, num=50):
+    """
+    Return evenly spaced numbers over a specified interval.
+
+    Returns `num` evenly spaced samples, calculated over the interval [start,
+    stop ].
+
+    The endpoint of the interval can optionally be excluded.
+
+    TODO: implement the `endpoint`, `restep`, and `dtype` keyword args
+    """
+    num = int(num)
+
+    blocksizes = _get_blocksizes(num, blocksize)
+
+    range_ = stop - start
+
+    space = float(range_) / (num - 1)
+
+    name = next(linspace_names)
+
+    dsk = {}
+    blockstart = start
+
+    for i, bs in enumerate(blocksizes):
+        blockstop = blockstart + ((bs - 1) * space)
+        task = (curry(np.linspace, num=bs), blockstart, blockstop)
+        blockstart = blockstart + (space * bs)
+        dsk[(name, i)] = task
+
+    return Array(dsk, name, blockdims=(blocksizes,))
+
+
+def arange(*args, **kwargs):
```

Comment:
I'm pretty much copying them from NumPy where applicable.  Is that okay? @mrocklin